rideshare_nguyen.rb

```diff
@@ -0,0 +1,153 @@
+  }
+}
+
+# Welcome statement of data report
+
+puts "Here is the Ride Share Data analysis from February 3rd, 2016 to February 5th, 2016: "
+puts "\n"
+
+########################################################
+# (1) Number of rides each driver has given
+# Based off of the "date_of_rides" array, finding the 
+# length of this array will give the total amount of rides
+
+# Strategy - first try to find amount for one driver first, 
+# then iterate. 
+# Came up with the following code for a single driver
+## puts "Driver 1 gave #{(rides[:DR0001][:date_of_ride]).length} rides"
+
+# To iterate the above code to give all drivers amount of rides
+# use the ".map" enumerable
+# to access the correct array, must include both key (driver) and value (data) 
+# as arguments for the block
+
+
+rides.map {|driver, data| 
+  puts "Driver #{driver} has given #{(data[:date_of_ride]).length} rides"
+}
+
+puts "\n"
+
+########################################################
+# (2) Earnings for each driver
+
+# Came up with the following code to calculate 
+# earnings for one driver:
+# ".inject(:+)" enumerable used to combine all elements in the 
+# array through addition, giving the total amount of 
+# $ a driver earned
+# puts "Driver 1 made: $#{(rides[:DR0001][:money_earned]).inject(:+)}"
+
+# When iterating, used same logic as (1), taking note of accessing the
+# proper value, ":money_earned"
+
+
+rides.map {|driver, data| 
+  puts "Driver #{driver} made a total of: $#{(data[:money_earned]).inject(:+)}"
+}
+
+puts "\n"
+
+########################################################
+# (3) Average rating for each driver
+# Came up with the following code to calculate 
+# average rating for one driver:
+# After accessing rating value array, first use ".inject(:+)"
+# to get the total ratings
+# Then divided by the length of the rating array
+# Then converted the total to a float
+# puts "Driver 3's average rating is 
+# #{(rides[:DR0003][:rating]).inject(:+) / (rides[:DR0003][:rating]).length.to_f}"
+
+# When iterating, wanted a nonrepeating decimal and used sprintf("%.2f", <average rating>)
+# to output only 2 decimal places
+
+rides.map {|driver, data| 
+  puts "Driver #{driver}'s average rating is: 
+  #{ sprintf("%.2f", ((data[:rating]).inject(:+) / ((data[:rating]).length).to_f))}"
+}
+
+puts "\n"
+
+########################################################
+# (4) Driver who made the most money
+# Add on to code from (2)
+# create new variable highest_earnings, set equal to zero
+# will use to hold as a placeholder and ultimately find the highest earning
+
+# Will also have variable highest_earner to hold the string
+# of the driver with the most earnings, for the final puts statement
+# Use .each to iterate and a conditional to compare current and previous earnings
+# replace highest_earnings value if current value is higher
+# print highest earner by using puts "#{driver}"
+
+highest_earnings = 0.0
+highest_earner = " "
+
+rides.each { |driver, data|
```

Comment:
You iterate over this hash a lot. Is there a way to keep this code organized while reducing the number of times you loop over `rides`? Perhaps methods could help.